libs/server/Resp/Parser/SessionParseState.cs

```diff
@@ -158,10 +182,26 @@ public void InitializeWithArguments(ArgSlice[] args)
 
             for (var i = 0; i < args.Length; i++)
             {
-                buffer[i] = args[i];
+                *(bufferPtr + i) = args[i];
             }
         }
 
+        /// <summary>
+        /// Limit access to the argument buffer to start at a specified index
+        /// and end after a specified number of arguments.
+        /// </summary>
+        /// <param name="idxOffset">Offset value to the underlying buffer</param>
+        /// <param name="count">Argument count (default: -1 for all remaining arguments)</param>
+        public SessionParseState Slice(int idxOffset, int count = -1)
```

Comment:
Use separate overloads to avoid `if` overhead?

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs#L357